sources/iTermAdvancedSettingsModel.m

```diff
@@ -628,6 +628,7 @@ static BOOL iTermAdvancedSettingsUseGCDUpdateTimerDefaultValue(void) {
 DEFINE_BOOL(openFileInNewWindows, NO, SECTION_WINDOWS @"Open files in new windows, not new tabs or split panes.\nThis affects shell scripts opened from Finder, for example. See also “Open files in split panes”.");
 DEFINE_BOOL(openFileInSplitPanes, NO, SECTION_WINDOWS @"Open files in split panes.\nThis affects shell scripts opened from Finder, for example. You must set “Open files in new windows, not new tabs” to “No” for this to take effect. See also “When opening a file in a split pane, should it be a vertical split?”");
 DEFINE_BOOL(openFileInVerticalSplitPane, YES, SECTION_WINDOWS @"When opening a file in a split pane, should it be a vertical split?")
+DEFINE_INT(newInstanceOpenStyle, 1, SECTION_WINDOWS @"How should new instances be opened when triggered by scripts or external applications?\n0: New Window\n1: New Tab (default)\n2: Vertical Split Pane\n3: Horizontal Split Pane\nThis provides a unified setting for controlling external instance creation.")
```

Comment:
The default should be out of range (-1 maybe?) to preserve compatibility with existing settings. As written, it would override them with "new tab". The other settings should be updated to say they are deprecated and direct the user to this one instead.